src/main/java/com/czertainly/core/api/cmp/error/CmpBaseException.java

```diff
@@ -0,0 +1,73 @@
+package com.czertainly.core.api.cmp.error;
+
+import com.czertainly.core.service.cmp.message.PkiMessageError;
+import org.bouncycastle.asn1.ASN1OctetString;
+import org.bouncycastle.asn1.DEROctetString;
+import org.bouncycastle.asn1.cmp.PKIBody;
+import org.bouncycastle.asn1.cmp.PKIFailureInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Exception for error handling for cmp failure scenarios.
+ * <p>
+ * citation (from rfc4210):

```

Comment:
Consolidate comment and names, for example, here is `rfc4210`, on other place, we have RFC 4210